### PR TITLE
Test server interface improvements

### PR DESCRIPTION
**Pull request by caspervonb**
- Renames util module to nats_server.
- Returns a result from run server helpers for cleaner stack traces in tests.
- Reorganizes/Rustifies the server helpers a bit.

### CODE REVIEW

**Review by derekcollison — COMMENTED** [review]
Why the Result and all of those unwraps? What does that buy us?

**Comment by caspervonb** [comment]
> Why the Result and all of those unwraps? What does that buy us?

Composability and stack traces pointing to the correct lines (which means quickfix list and things like that can jump to it).
If you panic somewhere deep in a utility your test failures are going to look like this:

```
failures:

---- two_connections stdout ----
thread 'two_connections' panicked at 'Oh no!', tests/nats_server/mod.rs:122:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- drop_flushes stdout ----
thread 'drop_flushes' panicked at 'Oh no!', tests/nats_server/mod.rs:122:5

---- async_subscription_drop stdout ----
thread 'async_subscription_drop' panicked at 'Oh no!', tests/nats_server/mod.rs:122:5
```

If you unwrap, that panics at the call site which is what you'd want in a test:

```
failures:

---- jetstream_publish stdout ----
thread 'jetstream_publish' panicked at 'Oh no!', tests/jetstream.rs:51:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

File: tests/auth_tls.rs [hunk 1]
```diff
@@ -1,12 +1,12 @@
 use std::io;
 use std::path::PathBuf;
 
-mod util;
-pub use util::*;
+mod nats_server;
+pub use nats_server::*;
 
 #[test]
 fn basic_tls() -> io::Result<()> {
```
Comment (Jarema) [inline tests/auth_tls.rs:8]: Remove the `Result` if we're using unwrap.

File: tests/drop.rs [hunk 2]
```diff
@@ -1,12 +1,12 @@
 use smol::future::FutureExt;
 use std::io;
 
-mod util;
-pub use util::*;
+mod nats_server;
+pub use nats_server::*;
 
 #[test]
 fn drop_flushes() -> io::Result<()> {
```
Comment (Jarema) [inline tests/drop.rs:8]: Remove the `Result` if we're using unwrap.

File: tests/drop.rs [hunk 3]
```diff
@@ -25,7 +25,7 @@ fn drop_flushes() -> io::Result<()> {
 
 #[test]
 fn two_connections() -> io::Result<()> {
```
Comment (Jarema) [inline tests/drop.rs:27]: Remove the `Result` if we're using unwrap.

File: tests/drop.rs [hunk 4]
```diff
@@ -41,7 +41,7 @@ fn two_connections() -> io::Result<()> {
 
 #[test]
 fn async_subscription_drop() -> io::Result<()> {
```
Comment (Jarema) [inline tests/drop.rs:43]: Remove the `Result` if we're using unwrap.

File: tests/jetstream.rs [hunk 5]
```diff
@@ -248,15 +251,20 @@ fn jetstream_publish() {
 
 #[test]
 fn jetstream_create_stream_and_consumer() -> io::Result<()> {
```
Comment (Jarema) [inline tests/jetstream.rs:253]: Remove the `Result` if we're using unwrap.

File: tests/jetstream.rs [hunk 6]
```diff
@@ -248,15 +251,20 @@ fn jetstream_publish() {
...
+    let nc = nats::connect(&s.client_url()).unwrap();
+    let js = nats::jetstream::new(nc.to_owned());
+
     js.add_stream("stream1")?;
     js.add_consumer("stream1", "consumer1")?;
     Ok(())
 }
 
 #[test]
 fn jetstream_queue_process() -> io::Result<()> {
```
Comment (Jarema) [inline tests/jetstream.rs:264]: Remove the `Result` if we're using unwrap.

File: tests/jetstream.rs [hunk 7]
```diff
@@ -294,7 +302,9 @@ fn jetstream_queue_process() -> io::Result<()> {
 
 #[test]
 fn jetstream_basics() -> io::Result<()> {
```
Comment (Jarema) [inline tests/jetstream.rs:304]: Remove the `Result` if we're using unwrap.

File: tests/auth_nkey.rs [hunk 8]
```diff
@@ -1,11 +1,11 @@
 use std::io;
 
-mod util;
-pub use util::*;
+mod nats_server;
+pub use nats_server::*;
 
 #[test]
 fn basic_nkey_auth() -> io::Result<()> {
-    let s = util::run_server("tests/configs/nkey.conf");
```
Comment (Jarema) [inline tests/auth_nkey.rs:8]: Remove the `Result` if we're using unwrap.

**Review by Jarema — CHANGES_REQUESTED** [review]

**Comment by caspervonb** [comment]
Became quite stale, faster to redo it.
Closing without merge.

**Pull request closed without merging**